parity-crypto/src/publickey/mod.rs

```diff
@@ -33,6 +33,14 @@ use lazy_static::lazy_static;
 pub use ethereum_types::{Address, Public};
 pub type Message = H256;
 
+use secp256k1::ThirtyTwoByteHash;
+pub struct ZeroesAllowedMessage(H256);
```

Comment:
does it have to be public?

Reply:
```suggestion
pub(super) struct ZeroesAllowedMessage(H256);
```

Reply:
or just move to where it's used?

Reply:
My thinking was that we'd construct this in `openethereum`; i.e. we'd make a breaking change in the API here, but in the end we didn't. As it stands it does not need to be `pub` at all I think.
If we do end up adding a new variant of `recover(…, message: &ZeroesAllowedMessage, …)` we'd need to export this so downstream can construct the param.